File: tests/fixtures/multi_lang_sample/sample.go
// Simple Go test file
package main

import (
	"fmt"
)

type Person struct {
	Name string
	Age  int
}

func NewPerson(name string, age int) *Person {
	return &Person{
		Name: name,
		Age:  age,
	}
}

func (p *Person) GetName() string {
	return p.Name
}

func (p *Person) SetName(name string) {
	p.Name = name
}

func (p *Person) GetAge() int {
	return p.Age
}

func Greet(name string) {
	fmt.Printf("Hello, %s!\n", name)
}

func Add(a, b int) int {
	return a + b
}

File: tests/fixtures/multi_lang_sample/sample.rs
// Simple Rust test file
use std::fmt;

pub struct Person {
    name: String,
    age: u32,
}

impl Person {
    pub fn new(name: String, age: u32) -> Self {
        Person { name, age }
    }
    
    pub fn get_name(&self) -> &str {
        &self.name
    }
    
    pub fn set_name(&mut self, name: String) {
        self.name = name;
    }
    
    pub fn get_age(&self) -> u32 {
        self.age
    }
}

pub fn greet(name: &str) {
    println!("Hello, {}!", name);
}

fn add(a: i32, b: i32) -> i32 {
    a + b
}
